scrapy_autoextract/providers.py

```diff
@@ -72,9 +77,7 @@ async def __call__(self):
 
         if "error" in data:
             self.stats.inc_value(f"autoextract/{page_type}/error/query")
-            raise QueryLevelError(
-                f"Error '{data['error']}' while processing {self.request}"
-            )
+            raise QueryError(data["query"], data["error"])
```

Comment:
What's you opinion on using scrapinghub-autoextract QueryError object instead (see https://github.com/scrapinghub/scrapinghub-autoextract/blob/9e7f3cc03208d54b858708b8f556acf7d312335b/autoextract/aio/errors.py#L60)?

Is it possible for two different types of QueryErrors to propagate to the user code? If it is, having them by the same name can be confusing.

Reply:
> What's you opinion on using scrapinghub-autoextract QueryError object instead

I don't think mixing up exceptions from two different libraries is a good idea. 

> Is it possible for two different types of QueryErrors to propagate to the user code?

No, QueryError exceptions on scrapinghub-autoextract are captured by the request method. When it happens, the latest results available are returned to the caller. A good idea is to turn this exception private to avoid confusion since it's not supposed to end-users.

Reply:
https://github.com/scrapinghub/scrapinghub-autoextract/issues/18